### PR TITLE
py-magenta: Submission with dependencies

### PR DESCRIPTION
**Pull request by essandess**
py-magenta: Submission with dependencies

* Depends upon: https://github.com/macports/macports-ports/pull/7840

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-sox/Portfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+checksums           rmd160  e6583059b2e46f28bc76895ccb65f34922fcfbf2 \
+                    sha256  176d3a1e541332a3b581fd60957f684069235fcf61881965f6c9e22e16797a85 \
+                    size    62438
+
+python.versions     37 38
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
```
Comment (mascguy) [inline python/py-sox/Portfile:26]: Steve, while this isn't a big deal, should py-sox declare a dependency on sox? (I realize this has already been committed, but perhaps for your next PR...)

Comment (essandess) [reply python/py-sox/Portfile:26]: Thanks for catching this. Fixed in #9858.

**Comment by harens** [comment]
Thank you very much @essandess for adding these ports. This is what allows MacPorts to grow and so I thank you for putting the time into writing these portfiles 👍.

I just wanted to mention that like others have said before, having PRs with large numbers of portfiles which are dependent on other PRs with large numbers of portfiles makes things quite difficult to review. 

In contrast to this, @mascguy's review of this PR https://github.com/macports/macports-ports/pull/7841#discussion_r566380988 lead to https://github.com/macports/macports-ports/pull/9858 where there's only one port, which makes things a lot easier and allows for small things to be picked up.

It would be great to have your work merged as soon as possible, since you put your time and effort into it, and this would help make it a bit easier. Thanks.

**Comment by essandess** [comment]
Believe it or not, these PRs are pared down given the huge missing downstream dependency tree for these ML frameworks. As some of the deeper branches are merged, I’ll try to break these into single PRs as much as possible.

> I just wanted to mention that like others have said before, having PRs with large numbers of portfiles which are dependent on other PRs with large numbers of portfiles makes things quite difficult to review.

**Pull request merged**